### PR TITLE
Spellcheck fixes

### PR DESCRIPTION
**Pull request by nhennigan**
Fixed typos across the docs and created more extensive custom_wordlist

### CODE REVIEW

File: docs/src/charm/howto/proxy.md
Comment (eaudetcobello) [inline docs/src/charm/howto/proxy.md:3]: there is also a lot of "e.g." around the code base

**Comment by nhennigan** [comment]
Removed spellcheck from automatically running on every PR. Will update with vale implementation later. I think we can still merge this once approved to stop the ever failing test and fix existing typos in the docs.

**Review by eaudetcobello — APPROVED** [review]
Nice work! Thanks

**Pull request merged**
